core/services/workflows/engine.go

```diff
@@ -934,8 +949,11 @@ func (e *Engine) executeStep(ctx context.Context, lggr logger.Logger, msg stepRe
 		},
 	}
 
-	e.metrics.incrementCapabilityInvocationCounter(ctx)
-	output, err := step.capability.Execute(ctx, tr)
```

Comment:
Execute is syncronous, so this should not be a problem?

Reply:
what do you mean?